rust/migrate-wicked/src/wireless.rs

```diff
@@ -416,27 +425,93 @@ where
     Ok(Some(Networks::deserialize(deserializer)?.network))
 }
 
-fn wireless_security_protocol(
-    wicked_value: &[String],
+fn all(array: &[String], needle: &[&str]) -> bool {
+    if array.is_empty() {
+        return false;
+    }
+
+    array.iter().all(|x| needle.iter().any(|&y| x == y))
+}
+
+fn any(array: &[String], needle: &[&str]) -> bool {
+    if array.is_empty() {
+        return false;
+    }
+
+    array.iter().any(|x| needle.iter().any(|&y| x == y))
+}
+
+fn guess_wireless_security_protocol(
+    network: &Network,
 ) -> Result<model::SecurityProtocol, anyhow::Error> {
-    if wicked_value.contains(&"wpa-psk".to_string())
-        || wicked_value.contains(&"wpa-psk-sha256".to_string())
-    {
+    let mgmt = &network.key_management;
+
+    let v = if any(mgmt, &["wpa-eap", "wpa-eap-sha256"]) {
+        Ok(model::SecurityProtocol::WPA2Enterprise)
+    } else if any(mgmt, &["wpa-eap-suite-b-192", "wpa-eap-suite-b"]) {
+        Ok(model::SecurityProtocol::WPA3Only)
+    } else if any(mgmt, &["wpa-psk", "wpa-psk-sha256"]) {
         Ok(model::SecurityProtocol::WPA2)
-    } else if wicked_value.contains(&"sae".to_string()) {
+    } else if any(mgmt, &["sae", "ft-sae"]) {
         Ok(model::SecurityProtocol::WPA3Personal)
-    } else if wicked_value.contains(&"wpa-eap".to_string())
-        || wicked_value.contains(&"wpa-eap-sha256".to_string())
-    {
-        Ok(model::SecurityProtocol::WPA2Enterprise)
-    } else if wicked_value.contains(&"owe".to_string()) {
+    } else if any(mgmt, &["owe"]) {
         Ok(model::SecurityProtocol::OWE)
-    } else if wicked_value.contains(&"wpa-eap-suite-b-192".to_string()) {
+    } else if any(mgmt, &["none"]) {
+        Ok(model::SecurityProtocol::WEP)
+    } else if network.wpa_eap.is_some() {
+        Ok(model::SecurityProtocol::WPA2Enterprise)
+    } else if network.wpa_psk.is_some() {
+        Ok(model::SecurityProtocol::WPA2)
+    } else {
+        Ok(model::SecurityProtocol::WEP)
+    };
+    log::warn!(
+        "Unsupported key-management protocol(s) '{}' guessing '{}'",
+        mgmt.join(","),
+        match v {
+            Ok(s) => s.to_string(),
+            _ => "-".to_string(),
+        }
+    );
+    v
```

Comment:
I think it's nicer to not wrap `v` (maybe also use a more descriptive name) in a `Result` and instead do this
```suggestion
    log::warn!(
        "Unsupported key-management protocol(s) '{}' guessing '{}'",
        mgmt.join(","),
        v
    );
    Ok(v)
```

Reply:
okay